miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/workflow/factory/BarcodableProgressStepFactory.java

```diff
@@ -0,0 +1,56 @@
+package uk.ac.bbsrc.tgac.miso.service.workflow.factory;
+
+import static uk.ac.bbsrc.tgac.miso.core.data.Barcodable.EntityType;
+import static uk.ac.bbsrc.tgac.miso.core.data.workflow.ProgressStep.InputType;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.view.BarcodableView;
+import uk.ac.bbsrc.tgac.miso.core.data.workflow.ProgressStep;
+import uk.ac.bbsrc.tgac.miso.core.data.workflow.impl.PoolProgressStep;
+import uk.ac.bbsrc.tgac.miso.service.BarcodableViewService;
+import uk.ac.bbsrc.tgac.miso.service.exception.ValidationError;
+import uk.ac.bbsrc.tgac.miso.service.exception.ValidationException;
+
+@Service
+@Transactional(rollbackFor = Exception.class)
+public class BarcodableProgressStepFactory implements ProgressStepFactory {
+  @Autowired
+  BarcodableViewService barcodableViewService;
+
+  @Override
+  public ProgressStep create(String input, Set<InputType> inputTypes) throws IOException {
+    List<BarcodableView> views = barcodableViewService.searchByBarcode(input, getEntityTypes(inputTypes));
+
+    if (views.size() == 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)